src/plots/cartesian/axes.js

```diff
@@ -1118,13 +1278,47 @@ axes.autoTicks = function(ax, roughDTick) {
             base = getBase(10);
             ax.dtick = roundDTick(roughDTick, base, roundBase10);
         }
+
+        if(isMinor) {
+            if(
+                typeof majorDtick === 'string' &&
+                majorDtick.charAt(0) === 'M'
+            ) {
+                if(majorDtick === 'M24') {
+                    ax.dtick = 'M12';
```

Comment:
@archmoj I love the new auto-minor algorithm! When you simply turn on minor ticks it's perfect as far as I can see 🏆  😍 

That said I think these date cases here are still a little too hard-coded if you stray from full auto mode. For example if I set `minor.nticks=12` or larger I'd expect year ticks to give one-month minor, 2-year ticks to give 2-month minor, etc. If I do that right now, 2-year ticks get 1-year minor, but when you zoom out to 5-year ticks you DROP to (the expected) 6-month minor ticks.